scripts/import-update-key

```diff
@@ -10,11 +10,35 @@ if [ -z "$keyfile" ]; then
   exit 1
 fi
 
-keyhash=$(gpg $gpgopts --with-fingerprint $keyfile | sed -ne 's#^pub  [^ ]\+/\([^ ]\+\).*#\1#p')
-if [ ${#keyhash} -ne 8 ]; then
+runver=$(gpg --version |grep -oP "gpg\s\(GnuPG\)\s\K.*") # runtime version
```

Comment:
direct use of gpg like this is not recommended https://www.gnupg.org/documentation/manuals/gnupg/Programmatic-use-of-GnuPG.html

for programmatic use `gpgme` is recommended, which is meant to have a stable interface.
See https://gnupg.org/documentation/manuals/gpgme/

Reply:
We already have `gpgme` and `pygpgme` in Dom0, perhaps using those we might be able to use the same code in XS8 and XS9 without having to perform version detection?

Reply:
Correct, especially the terminal user interface of GPG may change at any point, and for this reason, `yum` uses `pygpgme` on `el7`/XS8. Some historic background is here: https://wiki.python.org/moin/GnuPrivacyGuard
`pygpgme` is now part of `gpgme` and [under the new name `python3-gpg`](https://github.com/Intevation/intelmq-mailgen/issues/44), the `pgp` package is now a sub-package of `gpgme`:


https://packages.fedoraproject.org/pkgs/gpgme/python3-gpg/
https://packages.ubuntu.com/noble/python3-gpg

The pygpgme-0.3 used by `yum` of xs8 is too old for this example program to work:
https://python-guide.readthedocs.io/en/latest/scenarios/crypto/#gpgme-bindings
http://files.au.adversary.org/crypto/gpgme-python-howto.html#howto-get-key

Updating gnupg2 (would be good for known security reasons), gpgme and yum would solve that.

Reply:
We need to trust the imported key,
I did not find an easy way for gpgme to do this, which is compatible for XS8 and XS9.